app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgSendQueue.java

```diff
@@ -108,7 +109,8 @@ public static void handleNewBgReading(BgReading bgReading, String operation_type
                     context.sendBroadcast(updateIntent);
                 }
 
-                if (AppWidgetManager.getInstance(context).getAppWidgetIds(new ComponentName(context, xDripWidget.class)).length > 0) {
+                Context appContext = xdrip.getAppContext();//KS
```

Comment:
I think I needed this after I started calling BgSendQueue from WatchUpdateService.  For some reason, it didn't work as originally coded.  I will re-test this though, as I am now wondering maybe there could have been some other reason AppWidget was failing...